core/test/element_bin_util_test.cpp

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+
+#include "scipp/core/element/bin_util.h"
+
+#include <gtest/gtest.h>
+#include <vector>
+
+using namespace scipp;
+using namespace scipp::core::element;
+
+TEST(ElementBinUtilTest, begin_edge) {
+  constexpr auto check = [](const double coord, const scipp::index expected) {
+    const std::vector<double> edges{0, 1, 2, 3};
+    scipp::index bin{0};
+    scipp::index index;
+    begin_edge(bin, index, coord, edges);
+    EXPECT_EQ(index, expected);
+  };
+  check(-0.1, 0);
+  check(0.0, 0);
+  check(1.0, 1);
+  check(1.5, 1);
+  check(2.0, 2);
+  check(2.5, 2);
+  check(3.0, 2);
+}
+
+TEST(ElementBinUtilTest, end_edge) {
+  constexpr auto check = [](const double coord, const scipp::index expected) {
+    const std::vector<double> edges{0, 1, 2, 3};
+    scipp::index bin{0};
+    scipp::index index;
+    end_edge(bin, index, coord, edges);
+    EXPECT_EQ(index, expected);
+  };
+  check(-0.1, 2);
+  check(0.0, 2);
+  check(1.0, 2);
+  check(1.5, 3);
+  check(2.0, 3);
+  check(2.5, 4);
+  check(3.0, 4);
+}
```

Comment:
Missing tests for `subbin_sizes_exclusive_scan` ?

Reply:
Adding one in `variable/test/bin_util_test.cpp`. Testing the kernel in isolation is actually not so useful in this case.

Reply:
> Adding one in `variable/test/bin_util_test.cpp`. Testing the kernel in isolation is actually not so useful in this case.

The question is how useful it is to test such an implementation detail. Should maybe rather use the time to write more tests for `bin` itself, so even if we refactor the implementation later we do not waste the development time?

Reply:
I moved those function to a different file and namespace to make clear that they are implementation details.